### PR TITLE
Consumer proguard rules, javadoc artifact

### LINKED ISSUE #123: Include javadoc jar in releases
I was trying to view the Javadoc online at https://javadoc.io/doc/pub.devrel/easypermissions/, but it fails due to missing javadoc.jar.

### LINKED ISSUE #137: @AfterPermissionGranted should also be annotated with @Keep
if using proguard optimization, a @AfterPermissionGranted method could be completly inlined preventing EasyPermissions.onRequestPermissionsResult to find it.

    private static final int REQUEST_PERMISSION = 10;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        navigateToPlayer();
    }

    @Keep
    @AfterPermissionGranted(REQUEST_PERMISSION)
    private void navigateToPlayer() {
        if (EasyPermissions.hasPermissions(this, Manifest.permission.READ_EXTERNAL_STORAGE)) {
            Intent intent = new Intent(this, MusicPlayerActivity.class);
            startActivity(intent);
            finish();
        } else {
            EasyPermissions.requestPermissions(this, getString(R.string.read_permission_rational), REQUEST_PERMISSION, Manifest.permission.READ_EXTERNAL_STORAGE);
        }
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);

        // Forward results to EasyPermissions
        EasyPermissions.onRequestPermissionsResult(requestCode, permissions, grantResults, this);
    }

### PR DESCRIPTION
**Pull request by samtstern**
Fix #137
  Fix #123

### CODE REVIEW

File: easypermissions/proguard-rules.pro [hunk 1]
```diff
@@ -1,17 +1,3 @@
...
-
-# If your project uses WebView with JS, uncomment the following
-# and specify the fully qualified class name to the JavaScript interface
-# class:
-#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
-#   public *;
-#}
+-keepclassmembers class * {
+    @pub.devrel.easypermissions.AfterPermissionGranted <methods>;
+}
```
Comment (SUPERCILEX) [inline easypermissions/proguard-rules.pro]: nit: add a new line at the bottom of the file

File: build.gradle [hunk 2]
```diff
@@ -1,11 +1,11 @@
 buildscript {
     repositories {
         jcenter()
-        google()
+        maven { url 'https://maven.google.com' }
```
Comment (SUPERCILEX) [inline build.gradle]: @samtstern Any reason we added the more verbose version of the repo instead of keeping `google()`?

File: app/build.gradle [hunk 3]
```diff
@@ -16,7 +16,7 @@ android {
 
     buildTypes {
         release {
-            minifyEnabled false
+            minifyEnabled true
```
Comment (SUPERCILEX) [inline app/build.gradle]: I don't think we should deliver a proguarded version of the library. It personal drives me insane if I can't `Ctrl + B` a method and get the uncompiled/unobfuscated raw source code.

We have `minifyEnabled false` in FirebaseUI with a `consumerProguardFile` so the dev can decide to obfuscate their app if they want, but still develop it with the library source code.

Comment (samtstern) [inline app/build.gradle]: This is the app, not the library.

File: easypermissions/maven.gradle [hunk 4]
```diff
@@ -39,6 +39,18 @@ task sourcesJar(type: Jar) {
     classifier = 'sources'
 }
 
+task javadoc(type: Javadoc) {
```
Comment (SUPERCILEX) [inline easypermissions/maven.gradle]: I think we need to add [`failOnError = false`](https://docs.gradle.org/3.3/dsl/org.gradle.api.tasks.javadoc.Javadoc.html#org.gradle.api.tasks.javadoc.Javadoc:failOnError) for the build to work.

File: build.gradle (hunk 2 above)
Comment (samtstern) [inline build.gradle]: Because when I use `google()` my build fails and I didn't feel like figuring out why :-)

File: app/build.gradle (hunk 3 above)
Comment (SUPERCILEX) [inline app/build.gradle]: Oh lol. Oops! 😄

File: build.gradle (hunk 2 above)
Comment (SUPERCILEX) [inline build.gradle]: Haha, SGTM 😄

File: easypermissions/maven.gradle (hunk 4 above)
Comment (samtstern) [inline easypermissions/maven.gradle]: Yep! Gonna see if I can actually fix the errors first and then add this.

**Pull request merged**